### DISCUSSION TITLE
Automatic switch between Star Trails and NightLapse Video

### DISCUSSION THREAD

**Discussion by hwy0010** [gopro/labs#655] [Show and tell]
I have my GoPro 11 Hero Black stationary with USB powered to film lapse video, I have created a script to switch between Star Trails and NightLapse by sunrise/set. 
It uses the variable A to define which mode the camera should be, -1 for initial, 0 for NightLapse and 1 for Star Trails, In my code, I've set it to Star Trails after 1hr sunset, and back to NightLapse by sunrise. the visual script tool helps a lot. 
![image](https://github.com/gopro/labs/assets/12371612/0638e1e5-e676-4826-b756-efb476b1e11e)
code:
=A-1!D<A1!E+mNL+!S~!E+mST+!S>s3600<A1=A1+"Star Trails"+!R4>r<s3600>A1=A0+"NightLapse"+!R4!R28
https://gopro.github.io/labs/build/?id=lpnqpkwsj9ev27n3c

**Comment by dnewman-gpsw** [comment]
Very cool, thanks for sharing.  Got a video link to share its results?

Unfortunately you can't share a link to a build page (.../?id=xyz..), as all the build projects are cached by your browser, not on server. However you can share the PNG that is generated when saving the QR Code, that PNG has the build project embedded in its metadata. This is the currently only way to share projects.

**Reply by hwy0010** [reply]
Thanks for the heads up about sharing links. I will share the video when done.
I'm experiencing a problem about sunset/rise time, I'm thinking about whether "if current time > sunset + 1hrs" works, because I simply change the code from exact time like >18:00 to >s3600, and it don't seem to work properly, since I don't see anything about this in the doc, Can you tell me is it possible to do so? How can I make sure the camera gets the correct sunset time, can I do something like print sunset time? Thank you

**Reply by dnewman-gpsw** [reply]
No I hadn't thought of make sunrise/set times part of a conditional.  So >s3600 doesn't work.  It is a good idea, I will add that to backlog/todo list.
